### PR TITLE
Switch Aura to Babe consensus

### LINKED ISSUE #1303: Adjust chains names and ids to the multi-runtime realm before the next testnet restart
In order to be able to load the chain spec from a JSON file and ensure the client was compiled against the correct runtime, we need to stick to chains naming convention:
- chain ID must have a prefix corresponding to the runtime, e.g. `gear_dev`, `gear_local`, `vara_staging_testnet` etc.

For the meantime (as of 2022-08-10) we kept the current staging chain id intact and fiddled with the service code to attribute this name to the `gear` runtime even though it doesn't have a prefix to avoid another testnet restart from genesis only a week after the last restart. However, this has to be corrected sooner rather than later.

### PR DESCRIPTION
**Pull request by ukint-vs**
Since Vara runtime uses a Babe consensus, maintaining two different consensuses is not an easy task, it was decided to change the consensus on the test network.
Also with these changes we can use multiple runtimes in one binary.

Closes #1303 

@gear-tech/dev

### CODE REVIEW

**Review by ekovalev — APPROVED** [review]

File: node/service/src/chain_spec/gear.rs [hunk 1]
```diff
@@ -225,14 +238,27 @@ fn testnet_genesis(
...
+            authorities: Default::default(),
+            epoch_config: Some(gear_runtime::BABE_GENESIS_EPOCH_CONFIG),
         },
         grandpa: GrandpaConfig {
-            authorities: initial_authorities
+            authorities: Default::default(),
+        },
+        session: SessionConfig {
+            keys: initial_authorities
                 .iter()
```
Comment (breathx) [inline node/service/src/chain_spec/gear.rs:250]: into_iter to avoid cloning

File: runtime/gear/src/lib.rs [hunk 2]
```diff
@@ -89,12 +88,19 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     impl_name: create_runtime_str!("gear-node"),
     apis: RUNTIME_API_VERSIONS,
     authoring_version: 1,
-    spec_version: 1650,
+    spec_version: 1660,
```
Comment (breathx) [inline runtime/gear/src/lib.rs:91]: maybe reset spec version due to update to V3?

File: runtime/gear/src/lib.rs [hunk 3]
```diff
@@ -89,12 +88,19 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     impl_name: create_runtime_str!("gear-node"),
     apis: RUNTIME_API_VERSIONS,
     authoring_version: 1,
```
Comment (breathx) [inline runtime/gear/src/lib.rs:90]: We also can change ids in specs now to proper ones #1303 ?

File: utils/economic-checks/Cargo.toml [hunk 4]
```diff
@@ -59,6 +57,5 @@ lazy-pages = [
     "pallet-gear/lazy-pages",
 ]
 debug-wasm-mutate = []
-gear-native = ["gear-runtime", "sp-consensus-aura", "authoring-aura"]
+gear-native = ["gear-runtime", "sp-consensus-babe"]
 vara-native = ["vara-runtime", "sp-consensus-babe"]
```
Comment (breathx) [inline utils/economic-checks/Cargo.toml:61]: we can now remove optional usage of babe

File: utils/gear-runtime-test-cli/Cargo.toml [hunk 5]
```diff
@@ -44,6 +42,5 @@ gear-core-processor = { path = "../../core-processor" }
 junit-common = { path = "../junit-common" }
 
 [features]
-gear-native = ["gear-runtime", "sp-consensus-aura", "authoring-aura"]
+gear-native = ["gear-runtime", "sp-consensus-babe", "sp-finality-grandpa"]
```
Comment (breathx) [inline utils/gear-runtime-test-cli/Cargo.toml:45]: about optional also for finality-grandpa

**Review by breathx — APPROVED** [review]
👍

File: runtime/gear/src/lib.rs (hunk 2 above)
Comment (NikVolf) [reply runtime/gear/src/lib.rs:91]: better reset it in PR with json spec and such

File: runtime/gear/src/lib.rs (hunk 3 above)
Comment (ukint-vs) [reply runtime/gear/src/lib.rs:90]: done

File: utils/economic-checks/Cargo.toml (hunk 4 above)
Comment (ukint-vs) [reply utils/economic-checks/Cargo.toml]: done

File: .github/workflows/CD.yml [hunk 6]
```diff
@@ -1,8 +1,9 @@
 name: CD
 
 on:
```
Comment (breathx) [inline .github/workflows/CD.yml:3]: Won't empty `on` section trigger it on any activity?

File: node/service/src/chain_spec/gear.rs [hunk 7]
```diff
@@ -145,6 +150,8 @@ pub fn staging_testnet_config() -> Result<ChainSpec, String> {
                 // Initial PoA authorities
                 vec![
                     (
+                        hex!["c8e4df7eac6b52dc5281659f1f393903932ee4b1f69f311c3cb123bc40f9267a"]
```
Comment (breathx) [inline node/service/src/chain_spec/gear.rs:153]: Better to have some comment as it provided below

File: node/service/src/chain_spec/gear.rs [hunk 8]
```diff
@@ -153,6 +160,8 @@ pub fn staging_testnet_config() -> Result<ChainSpec, String> {
                             .unchecked_into(),
                     ),
                     (
+                        hex!["3c4c519e3d7149c93181e8e3762562db6f580c27502e9a6ab2f7464d6185241b"]
```
Comment (breathx) [inline node/service/src/chain_spec/gear.rs:163]: Same for this one

File: node/service/src/chain_spec/gear.rs [hunk 9]
```diff
@@ -161,6 +170,8 @@ pub fn staging_testnet_config() -> Result<ChainSpec, String> {
                             .unchecked_into(),
                     ),
                     (
+                        hex!["587e919f8149e31f7d4e99e8fbdf30ff119593376f066e20dacda9054892b478"]
```
Comment (breathx) [inline node/service/src/chain_spec/gear.rs:173]: This

File: node/service/src/chain_spec/gear.rs [hunk 10]
```diff
@@ -169,6 +180,8 @@ pub fn staging_testnet_config() -> Result<ChainSpec, String> {
                             .unchecked_into(),
                     ),
                     (
+                        hex!["f2fd6936b8ddad025d329ff2d6b5577e6381cb25333f6f17f592494b0b61ef55"]
```
Comment (breathx) [inline node/service/src/chain_spec/gear.rs:183]: And this also

File: node/service/src/chain_spec/gear.rs [hunk 11]
```diff
@@ -225,14 +238,27 @@ fn testnet_genesis(
...
+        },
+        session: SessionConfig {
+            keys: initial_authorities
+                .into_iter()
+                .map(|x| {
+                    (
+                        x.0.clone(),
+                        x.0.clone(),
+                        SessionKeys {
+                            babe: x.1.clone(),
```
Comment (breathx) [inline node/service/src/chain_spec/gear.rs:256]:
```suggestion
                        x.0,
                        SessionKeys {
                            babe: x.1,
```

File: node/service/src/chain_spec/vara.rs [hunk 12]
```diff
@@ -247,14 +247,14 @@ fn testnet_genesis(
...
         session: SessionConfig {
             keys: initial_authorities
-                .iter()
+                .into_iter()
                 .map(|x| {
                     (
                         x.0.clone(),
                         x.0.clone(),
                         SessionKeys {
                             babe: x.1.clone(),
```
Comment (breathx) [inline node/service/src/chain_spec/vara.rs:256]:
```suggestion
                        x.0,
                        SessionKeys {
                            babe: x.1,
```

**Review by breathx — APPROVED** [review]
Great

File: .github/workflows/CD.yml (hunk 6 above)
Comment (ukint-vs) [reply .github/workflows/CD.yml:3]: `workflow_dispatch` here

**Pull request merged**